src/commands/Vatsim.ts

```diff
@@ -0,0 +1,233 @@
+import {Category, NotBot} from "@discordx/utilities";
+import {AutocompleteInteraction, CommandInteraction, MessageEmbed} from "discord.js";
+import {Client, Discord, Guard, Slash, SlashChoice, SlashOption} from "discordx";
+import {injectable} from "tsyringe";
+
+import {GuildOnly} from "../guards/GuildOnly.js";
+import {RequiredBotPerms} from "../guards/RequiredBotPerms.js";
+import {VatsimManager} from "../model/framework/manager/VatsimManager.js";
+import type {VatsimAtc, VatsimAtis, VatsimPilot} from "../model/Typeings.js";
+import logger from "../utils/LoggerFactory.js";
+import {InteractionUtils, ObjectUtil} from "../utils/Utils.js";
+
+@Discord()
+@Category("VATSIM commands")
+@injectable()
+export class Vatsim {
+    public constructor(private _ivaoManager: VatsimManager) {
+
+    }
+
+    @Slash("vatsim", {
+        description: "Gives you the information for the chosen call sign on the VATSIM network"
+    })
+    @Guard(NotBot, RequiredBotPerms({
+        textChannel: ["EMBED_LINKS"]
+    }), GuildOnly)
+    private vatsim(
+        @SlashChoice("atc", "pilot")
+        @SlashOption("type", {
+            // TODO: fill this info out
+            description: "Supply a description",
+            type: "STRING",
+            required: true
+        })
+            type: "atc" | "pilot",
+        @SlashOption("call-sign", {
+            description: "What call sign would you like the bot to give information for?",
+            autocomplete: (interaction: AutocompleteInteraction) => InteractionUtils.search(interaction, VatsimManager),
+            type: "STRING",
+            required: true
+        })
+            callSign: string,
+        interaction: CommandInteraction,
+        client: Client
+    ): Promise<void> {
+        const vatsimEmbed = new MessageEmbed()
+            .setTitle(`${callSign.toUpperCase()}`)
+            .setColor("#0099ff")
+            .setFooter({
+                text: `${client.user.username} • This is not a source for official briefing • Please use the appropriate forums • Source: VATSIM API`
+            })
+            .setTimestamp();
+
+        try {
+            let vatsimClient = this._ivaoManager.getClientInfo(callSign, type) as VatsimPilot | VatsimAtc;
+            vatsimEmbed.setTitle(`Vatsim : ${callSign}`);
+            vatsimEmbed.addFields({
+                    name: "Call Sign",
+                    value: vatsimClient.callsign,
+                    inline: true
+                },
+                {
+                    name: "CID",
+                    value: vatsimClient.cid.toString(),
+                    inline: true
+                },
+                {
+                    name: "Name",
+                    value: vatsimClient.name,
+                    inline: true
+                });
```

Comment:
I think we can similarly move few common fields out in IVAO also

Reply:
odd, I did this before.
anyhow, done!